### DISCUSSION TITLE
How to sign a message hash

### DISCUSSION THREAD

**Discussion by LewisPringle** [rainbow-me/rainbowkit#1171] [Q&A]
I am trying to create a signing component in another library and integrate with rainbowkit/wagami (as the wallet connector).

That library has an abstract signer API that just does sign(messageHash) (it has higher level classes elesewhere to do the message digest stuff).

But the WAGAMI api for class Signer - just has signMessage and signTransaction. Nothing obvious to sign just the message hash itself.

**Comment by albertdugba** [comment]
Hey pal @LewisPringle  did you find a solution to this ?

**Comment by LewisPringle** [comment]
yes, I found a solution, but it was so long ago, I've forgotten most of the details. Mostly I ended up using different libraries and re-implementing all the logic for message signing on top abstractly, so I could vector to either the wagmi api (I think signtransaction) - and when using a different signer, it would implement that itself via digesting and signing itself (signing the message hash).

The most useful think I discovered - I don't have access to anymore (since done for another company) - is what libraries I ended up using that worked well to support this fancier signing logic. Sorry - I guess not super helpful.
